### PR TITLE
On pay now redirected users to checkout page instead of order summary for non FA programs

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/3105

#### What's this PR do?
- Redirect user to checkout instead of progress summary for `NON FA` programs. 

#### How should this be manually tested?
- Click `pay now` buttons for no fa course

@pdpinch

### CODE REVIEW

**Comment by noisecapella** [comment]
It looks like the price is still part of the modal. The linked issue says it should be removed
screenshot from 2017-05-12 10-55-13

File: static/js/containers/DashboardPage.js [hunk 1]
```diff
@@ -550,6 +552,11 @@ class DashboardPage extends React.Component {
     );
   }
 
+  dispatchCheckout = (courseId: string) => {
+    const { dispatch } = this.props;
```
Comment (noisecapella) [inline static/js/containers/DashboardPage.js]: Can you add a test which would cover this function?

File: static/js/components/dashboard/CourseRow.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ export default class CourseRow extends React.Component {
     setEnrollSelectedCourseRun: (r: CourseRun) => void,
     setEnrollCourseDialogVisibility: (b: boolean) => void,
     ui: UIState,
+    checkout: Function,
```
Comment (noisecapella) [inline static/js/components/dashboard/CourseRow.js]: Can you make this type more specific?

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella

File: static/js/containers/DashboardPage_test.js [hunk 3]
```diff
@@ -619,4 +621,40 @@ describe('DashboardPage', () => {
       });
     });
   });
+
+  describe("checkout", () => {
+    let dashboardResponse;
+    beforeEach(() => {
+      // Limit the dashboard response to 1 program
+      dashboardResponse = {"programs": [R.clone(DASHBOARD_RESPONSE.programs[0])]};
```
Comment (noisecapella) [inline static/js/containers/DashboardPage_test.js]: Can you use `makeProgram()` instead? This should also let you use `program.courses[0]` for the course in the test below.

File: static/js/containers/DashboardPage_test.js [hunk 4]
```diff
@@ -619,4 +621,40 @@ describe('DashboardPage', () => {
...
+    });
+
+    it('redirects to edX when the checkout API tells us to', () => {
+      let promise = Promise.resolve(EDX_CHECKOUT_RESPONSE);
+      let checkoutStub = helper.sandbox.stub(storeActions, 'checkout').returns(() => promise);
+      let course = makeCourse();
+      // Set all course runs to unpaid
+      course.runs = R.chain(R.set(R.lensProp('has_paid'), false), course.runs);
```
Comment (noisecapella) [inline static/js/containers/DashboardPage_test.js]: `course` can be modified directly so you shouldn't need the lens or setter methods here, and it looks like you only need to modify one run in the course. Can you do something like: `course.runs[0] = { ...course.runs[0], status: STATUS_CAN_UPGRADE, course_end_date: moment().add(-1, 'months'), ` and so on

File: static/js/containers/LearnerSearchPage_test.js [hunk 5]
```diff
@@ -230,7 +230,7 @@ describe('LearnerSearchPage', function () {
         query: 'xyz',
         type: 'phrase_prefix',
       });
-      assert.equal(window.location.toString(), "http://fake/?q=xyz");
+      assert.equal(window.location.toString(), "http://edx.org/?q=xyz");
```
Comment (noisecapella) [inline static/js/containers/LearnerSearchPage_test.js]: Why is this change necessary? If the problem is that `window.location` from one test is affecting this one, we should solve the problem in `babelhook.js` or `global_init.js` instead, where we handle `window.location` mocking

File: static/js/containers/DashboardPage_test.js (hunk 3 above)
Comment (amir-qayyum-khan) [inline static/js/containers/DashboardPage_test.js]: dashboard is not rendering with makeProgram(), i dont know what is missing

File: static/js/containers/LearnerSearchPage_test.js (hunk 5 above)
Comment (amir-qayyum-khan) [inline static/js/containers/LearnerSearchPage_test.js]: any thing in ur mind

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella
